### PR TITLE
kzs_doc: Implemented 'doc_type' option check

### PR DESCRIPTION
**Pull request by sergey-safarov**
now call like 'kz_datamgr:open_cache_doc(AccountDb, {<<"user">>, ConfigId})' works properly

### CODE REVIEW

File: core/kazoo_data/src/kzs_doc.erl [hunk 1]
```diff
@@ -42,16 +42,33 @@ open_doc(#{server := {App, Conn}}, DbName, DocId, Options) ->
...
+            lager:info("denying access to soft deleted doc: ~s", [kz_doc:id(Doc)]),
+            {'error', 'not_found'}
+    end.
+
+-spec maybe_type_mismatch(kz_json:object(), kz_term:api_ne_binary()) ->
+          {'ok', kz_json:object()} |
+          {'error', 'not_found'}.
+maybe_type_mismatch(Doc, 'undefined') ->
+    {'ok', Doc};
+maybe_type_mismatch(Doc, ReqestedType) ->
```
Comment (jamesaimonetti) [inline core/kazoo_data/src/kzs_doc.erl]: s/Reqested/Requested

Comment (sergey-safarov) [reply core/kazoo_data/src/kzs_doc.erl]: fixed

**Comment by lazedo** [comment]
@sergey-safarov trying to understand this pr. you say this pr fixes `now call like 'kz_datamgr:open_cache_doc(AccountDb, {<<"user">>, ConfigId})' works properly` but i don't see a relation here. also, this check is already in crossbar. please provide more context on this, maybe with an example that's not working as you expected to.

**Comment by sergey-safarov** [comment]
I want be sure about document type fetched from database.
For this purpose exist function [`kz_datamgr:open_cache_doc/2`](https://github.com/2600hz/kazoo/blob/master/core/kazoo_data/src/kz_datamgr.erl#L533-L534).

Functions defines `doc_type` options but implementation for this check not exist.
PR add  `doc_type` options check then document fetched from database.

**Comment by lazedo** [comment]
@sergey-safarov implementation is not intended here. the `doc_type` is to help resolving dataplan access. `crossbar` does what you to put here.

**Comment by sergey-safarov** [comment]
I not want create own doc_type with own `kzd` module.
Inside this module i want be sure doc_type 'opened using `kz_datamgr:open_cache_doc(AccountDb, {<<"my_doc_type">>, ConfigId})`
So crosbar here not applicable.
> doc_type is to help resolving dataplan access

Could give me reference for such usage.

**Comment by lazedo** [comment]
> I not want create own doc_type with own `kzd` module.

not saying that you do. what i'm saying is that this `PR` is not acceptable, you're introducing unwanted `atm` changes in the core data access library.

if you're not using `crossbar` then you can do that in your library `foo:open_cache_doc/3` that will call `kz_datamgr:open_cache_doc /3` or other, and can do any pre-process or post-process

**Comment by lazedo** [comment]
@sergey-safarov sorry for closing this but we need to move forward and merge/close pending prs as we approach 5.x

**Pull request closed without merging**